### PR TITLE
Bug 955999 - Running webapp-zip.js on node.js

### PR DESCRIPTION
**Pull request by rickychien**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[RickyChien Ricky Chien (RickyChien)](https://github.com/RickyChien) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f93fd3920c21)

**Comment by try-server-hook** [comment]
[RickyChien Ricky Chien (RickyChien)](https://github.com/RickyChien) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=77d900d792ee)

**Comment by try-server-hook** [comment]
[RickyChien Ricky Chien (RickyChien)](https://github.com/RickyChien) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e41db772a9cb)

File: build/utils-xpc.js [hunk 1]
```diff
@@ -1121,10 +1119,8 @@ function addEntryContentWithTime(zip, pathInZip, data, time, compression) {
     input.init(data, -1, -1, 0);
   }
 
-  zip.addEntryStream(
-    pathInZip, time || 0, compression, input, false);
+  zip.addEntryStream(pathInZip, Date.now() * 1000, compression, input, false);
```
Comment (cctuan) [inline build/utils-xpc.js]: the modification time of the entry in microseconds, so I think we don't need * 1000.

File: build/utils-node.js [hunk 2]
```diff
@@ -459,8 +459,15 @@ module.exports = {
...
 
+  hasFileInZip: function(zip, pathInZip) {
+    return zip.file(pathInZip);
+  },
+
   closeZip: function(zip, zipPath) {
-    fs.writeFileSync(zipPath, zip.generate({ type: 'nodebuffer' }));
+    fs.writeFileSync(zipPath, zip.generate({
+      type: 'nodebuffer',
+      platform: process.platform
```
Comment (cctuan) [inline build/utils-node.js]: I guess you refer to below link.
https://github.com/Stuk/jszip/issues/194
It's no written in official document, so I append the link here.

File: build/utils-xpc.js (hunk 1 above)
Comment (RickyChien) [inline build/utils-xpc.js]: Yeah I think so. but in fact I got correct time after * 1000

**Comment by try-server-hook** [comment]
[RickyChien Ricky Chien (RickyChien)](https://github.com/RickyChien) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d39fdd190b19)

**Pull request merged**